### DISCUSSION TITLE
From a non-component class, should I expect a Modal dialog to appear when I call  `IModalService.Show<>`?

### DISCUSSION THREAD

**Discussion by szalapski** [Blazored/Modal#457] [Q&A] [answered]
Ordinarliy, a Modal dialog appears when I call `IModalService.Show<>` from a component, when IModalService is injected, which works great.

Should I expect a Modal dialog to appear when I call  `IModalService.Show<>` from a method in a plain old C# class that is injected into a component (where IModalService is of course injected via the constructor)?  The method is called from the component, of course.

I couldn't imagine why it wouldn't--after all, the runtime is injecting the same IModalService into both, right?

```c#
public class Class1 : IClass1
{
    public Class1(IModalService modal)
    {
        Modal = modal ?? throw new ArgumentNullException(nameof(modal));
    }
    protected IModalService Modal { get; set; }

    public void Show()
    {
        Modal.Show<MyDialog>();
    }
}
```

Then, from the component:

```
@inject IClass1 class1;
@inject IModalService modal;
<button @onclick="class1.Show" >Show it</button> @* doesn't work *@
<button @onclick="@() => modal.Show<MyDialog>()" >Show it</button> @* works *@
```

CC @chrissainty

**Comment by szalapski** [comment, marked as answer]
Ah, When I use "AddScoped" or "AddTransient" in my new class, it works.  But a registering the class as a Singleton causes the trouble.  Any idea why?

**Reply by chrissainty** [reply]
I would assume it's because the Blazored Modal service is registered as Scoped and you can't inject scoped services into Singletons. I'm surprised you're not seeing an error due to that.

**Reply by szalapski** [reply]
Ah, yes, that makes perfect sense.  Yes, I wonder why an error would not be evident?  It just fails silently.
